gtsam/discrete/DiscreteFactorGraph.cpp

```diff
@@ -210,6 +210,12 @@ namespace gtsam {
     for (auto&& factor : factors) product = (*factor) * product;
     gttoc(product);
 
+    // Sum all the potentials by pretending all keys are frontal:
+    auto normalization = product.sum(product.size());
```

Comment:
Wondering whether we could use `max` here to avoid the impression the factor needs to be normalized…